lib/systems/src/MainChargeSystem.cpp

```diff
@@ -3,19 +3,26 @@
 #include <algorithm>
 #include <cmath>
 
-MainChargeSystem::MainChargeSystem(float target_volt, float max_charge_current, float max_cell_temp) : target_voltage(target_volt), max_ac_charging_current(max_charge_current), max_allowable_cell_temperature(max_cell_temp){}
+/**
+ * @brief void returning function that sets current_setpoint in the output struct to a calculated current value based on current cell voltage and temp 
+ * @param ACUAllData_s struct populated with raw data from ACU
+ * @param CCUParams struct with params that control the charge control unit
+ */
+float MainChargeSystem::calculate_charge_current(ACUAllData_s inputValues, CCUParams &chargeParams)
+ {
 
-void MainChargeSystem::charge_cycle(CCUInput_s inputValues, CCUOutput_s &outputValues){
-    if (inputValues.max_cell_voltage >= target_voltage || inputValues.max_cell_temperature > max_allowable_cell_temperature) { //checks whether or not any of the cells are charged yet
-      outputValues.current_flow = false; //this will be function that actually turns on/off current flow 
-      outputValues.current_setpoint = 0;
+    double max_cell_voltage = *std::max_element(inputValues.voltages, inputValues.voltages + (sizeof(inputValues.voltages) / sizeof(inputValues.voltages[0]))); //sizeof is needed because max value takes in a start and end memory location of the array
+    double max_cell_temperature = *std::max_element(inputValues.cell_temperatures, inputValues.cell_temperatures + (sizeof(inputValues.cell_temperatures) / sizeof(inputValues.cell_temperatures[0])));
+
+    if (max_cell_voltage >= chargeParams.target_voltage || max_cell_temperature > chargeParams.max_allowable_cell_temperature) { //checks whether or not any of the cells are charged yet
```

Comment:
nit: comment is inaccurate. it's not that any of the cells are charged, it's that one of the exit conditions (overvoltage or over temp) has been met.